libindy/src/commands/non_secrets.rs

```diff
@@ -161,7 +168,11 @@ impl NonSecretsCommandExecutor {
 
         self._check_type(type_)?;
 
-        let res = self.wallet_service.update_record_tags(wallet_handle, type_, id, tags_json)?;
+        let tags: HashMap<String, String> = match serde_json::from_str(tags_json) {
+            Ok(tag_names) => tag_names,
+            Err(serde_json_err) => return Err(IndyError::WalletError(WalletError::InputError(format!("Invalid tags input: {}", tags_json))))
+        };
+        let res = self.wallet_service.update_record_tags(wallet_handle, type_, id, &tags)?;
```

Comment:
match is not required here and several places below
let tags: <HashMap<String, String>> = serde_json::from_str(tags_json)
            .map_err(|err| CommonError::InvalidStructure(format!("Cannot deserialize tags: {:?}", err)))?;